shell/platform/darwin/ios/framework/Source/FlutterKeyboardManagerTest.mm

```diff
@@ -17,364 +15,253 @@
 
 FLUTTER_ASSERT_ARC;
 
-namespace flutter {
-class PointerDataPacket {};
-}  // namespace flutter
-
 using namespace flutter::testing;
 
-/// Sometimes we have to use a custom mock to avoid retain cycles in ocmock.
-@interface FlutterEnginePartialMock : FlutterEngine
-@property(nonatomic, strong) FlutterBasicMessageChannel* lifecycleChannel;
-@property(nonatomic, weak) FlutterViewController* viewController;
-@property(nonatomic, assign) BOOL didCallNotifyLowMemory;
-@end
-
-@interface FlutterEngine ()
-- (BOOL)createShell:(NSString*)entrypoint
-         libraryURI:(NSString*)libraryURI
-       initialRoute:(NSString*)initialRoute;
-- (void)dispatchPointerDataPacket:(std::unique_ptr<flutter::PointerDataPacket>)packet;
-@end
-
-@interface FlutterEngine (TestLowMemory)
-- (void)notifyLowMemory;
-@end
-
-extern NSNotificationName const FlutterViewControllerWillDealloc;
-
-/// A simple mock class for FlutterEngine.
-///
-/// OCMockClass can't be used for FlutterEngine sometimes because OCMock retains arguments to
-/// invocations and since the init for FlutterViewController calls a method on the
-/// FlutterEngine it creates a retain cycle that stops us from testing behaviors related to
-/// deleting FlutterViewControllers.
-@interface MockEngine : NSObject
-@end
-
-@interface FlutterKeyboardManagerUnittestsObjC : NSObject
-- (bool)nextResponderShouldThrowOnPressesEnded;
-- (bool)singlePrimaryResponder;
-- (bool)doublePrimaryResponder;
-- (bool)singleSecondaryResponder;
-- (bool)emptyNextResponder;
-@end
-
-namespace {
-
-typedef void (^KeyCallbackSetter)(FlutterUIPressProxy* press, FlutterAsyncKeyCallback callback)
-    API_AVAILABLE(ios(13.4));
-typedef BOOL (^BoolGetter)();
-
-}  // namespace
-
+// These tests were designed to run on iOS 13.4 or later.
+API_AVAILABLE(ios(13.4))
```

Comment:
I think by putting the availability on the class `FlutterKeyboardManagerTest` it won't need to be checked in setUp or decorated on the methods.